PowerShell.Common.props

```diff
@@ -1,10 +1,100 @@
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <!--
+        We should ~explicitly~ assign versions properties in the target
+        because:
+          1. By default the version properties defined in global PropertyGroup (not target!) in 'Microsoft.NET.DefaultAssemblyInfo.targets'
+          2. Properties in global 'PropertyGroups' is assigned before any targets
+          3. and cannot be conditionally excecuted as targets (after this target).
+
+        To assign vertions properties we should excecute the target:
+          '_GenerateRestoreProjectSpec' - before 'Restore' target
+          'GenerateNuspec'              - before 'Pack' target
+          'BeforeBuild'                 - before 'Build' target
+  -->
+  <Target Name="GetPSCoreVersionFromGit"
+    BeforeTargets="_GenerateRestoreProjectSpec;GenerateNuspec;BeforeBuild"
+  >
+
+    <PropertyGroup>
+      <GitExe>git</GitExe>
+    </PropertyGroup>
+
+    <Exec Command='$(GitExe) rev-parse --show-cdup' ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="GitCDUP" />
+    </Exec>
+
+    <PropertyGroup>
+      <GitInfoBaseDir Condition="'$(GitInfoBaseDir)' == ''">$(MSBuildProjectDirectory)/$(GitCDUP)/.git</GitInfoBaseDir>
+    </PropertyGroup>
+
+    <Exec Command='$(GitExe) --git-dir="$(GitInfoBaseDir)" describe --abbrev=60 --long' ConsoleToMSBuild="true" EchoOff="true">
```

Comment:
I would suggest assigned the git directory externally.  The reason I've added it to all the git  commands I've touch is that sometimes git cannot find the directory, which the current solution will not solve.